service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ServiceClient.java

```diff
@@ -65,11 +109,140 @@ public static ServiceClient createFromConnectionString(String connectionString,
             throw new IllegalArgumentException("ServiceClientOptions cannot be null for this constructor");
         }
 
-        // Codes_SRS_SERVICE_SDK_JAVA_SERVICECLIENT_12_002: [The constructor shall create IotHubConnectionString object using the IotHubConnectionStringBuilder]
-        IotHubConnectionString iotHubConnectionString = IotHubConnectionStringBuilder.createConnectionString(connectionString);
+        IotHubConnectionString iotHubConnectionString = IotHubConnectionStringBuilder.createIotHubConnectionString(connectionString);
+
+        this.hostName = iotHubConnectionString.hostName;
+        this.userName = iotHubConnectionString.getUserString();
+        this.sasToken = new IotHubServiceSasToken(iotHubConnectionString).toString();
+        this.iotHubServiceClientProtocol = iotHubServiceClientProtocol;
+        this.options = options;
+        this.amqpMessageSender =
+                new AmqpSend(
+                        this.hostName,
+                        this.userName,
+                        this.sasToken,
```

Comment:
if we calculate the sas token during client creation, how will we refresh them?

Reply:
This constructor doesn't support renewing SAS tokens yet. This feature work won't change that, but it will add token refresh for the new constructors that take TokenCredential or AzureSasCredential